### ISSUE TITLE
Make case sensitivity consistent

### ISSUE THREAD

**Issue by iwahbe** [pulumi/pulumi-yaml#355]
## Hello!

- Vote on this issue by adding a 👍 reaction
- If you want to implement this feature, comment to let us know (we'll work with you on design, scheduling, etc.)

### The Problem
Pulumi YAML is sometimes case sensitive, sometimes not. When it is case sensitive, different capitalization schemes are required. We need to decide on a consistent capitalization scheme. Any decision is better than what we have now.

Our decision is complicated by the fact that Pulumi YAML consumes certain fields, but passes through others. Consumed fields can be cross–checked against the relevant schema to correct for capitalization. Fields that are passed through must be case sensitive.
This short program gives an example of the problem:
<img width="478" alt="Screen Shot 2022-10-07 at 2 36 37 PM" src="https://user-images.githubusercontent.com/22222529/194661708-4a0f5488-a84d-4aa8-b76f-d1d98e3a1d07.png">

This is not intuitive right now, but it definitely should be.
### The Solution
TLDR: The least confusing option is to make everything case sensitive.

We must make certain areas case sensitive, such as string values and the keys to maps. The values are entirely pass through, and there is no way to correct invalid casing. Because of the uniform syntax of YAML, it's not always clear when you are in a case sensitive setting. We can eliminate this question by teaching users that the answer is always: yes, it is case sensitive. This will improve the writability of Pulumi YAML.

We currently require both CamelCase and camelCase styles when interacting with functions and properties respectively. Other areas, such as Pulumi YAML top level keys, give the user the ability to choose which style they prefer. We have learned recently that it is beneficial for the language to enforce a preferred naming style (Rust, Go) because it forces similar programs to look similar. We should adopt this best practice to improve the readability of Pulumi YAML.

Under this solution, we would standardize all Pulumi YAML keys and values on camelCase, which matches with the property names in our schemas.

This would be a breaking change for our existing ecosystem. We would roll out a release that generates warnings for invalid casing, and then another release that would turn these warnings into errors.

### Affected area/feature

The `ast` and `syntax` modules.
